prow/cmd/phaino/local.go

```diff
@@ -246,7 +250,87 @@ func convertToLocal(ctx context.Context, log *logrus.Entry, pj prowapi.ProwJob,
 		log.Warn("Ignoring resource requirements")
 	}
 
+	// Setup gcloud credentials
+	if opt.useLocalGcloudCredentials || envExists(container.Env, gcloudCredEnvName) {
+		gcloudKey := os.Getenv(gcloudCredEnvName)
+		// If GOOGLE_APPLICATION_CREDENTIALS is not empty, also mount the key file
+		// to the container
+		if gcloudKey != "" {
+			if _, err := os.Stat(gcloudKey); !os.IsNotExist(err) {
+				localArgs = append(localArgs, "-e", gcloudCredEnvName+"="+gcloudKey)
+				localArgs = append(localArgs, "-v", gcloudKey+":"+gcloudKey+":ro")
+			} else {
+				logrus.Warning("The GOOGLE_APPLICATION_CREDENTIALS file does not exist on your local machine, thus gloud authentication won't work in the container")
```

Comment:
I think it would be cleaner to just print a general warning saying "failed using default gcloud credential" after the outer if condition

Reply:
I don't quite get what you meant for this comment so I'm leaving the logic unchanged. Let's chat about it later.